golem-cli/src/command/component.rs

```diff
@@ -78,6 +100,32 @@ pub enum ComponentSubCommand<ProjectRef: clap::Args, ComponentRef: clap::Args> {
         non_interactive: bool,
     },
 
+    /// Updates an existing component by uploading a new version of its WASM and functions.
+    /// All parameters are taken from the app manifest.
+    #[command()]
+    UpdateWithManifest {
```

Comment:
I'm not sure what your approach to backwards compat is, so I added this and update as new commands for now.